design/Extend-VolumePolicies-to-support-more-actions.md

```diff
@@ -0,0 +1,176 @@
+# Extend VolumePolicies to support more actions
+
+## Abstract
+
+Currently, the [VolumePolicies feature](https://github.com/vmware-tanzu/velero/blob/main/design/Implemented/handle-backup-of-volumes-by-resources-filters.md) which can be used to filter/handle volumes during backup only supports the skip action on matching conditions. Users need more actions to be supported.
+
+## Background
+
+The `VolumePolicies` feature was introduced in Velero 1.11 as a flexible way to handle volumes. The main agenda of
+introducing the VolumePolicies feature was to improve the overall user experience when performing backup operations
+for volume resources, the feature enables users to group volumes according the `conditions` (criteria) specified and
+also lets you specify the `action` that velero needs to take for these grouped volumes during the backup operation.
+The limitation being that currently `VolumePolicies` only supports `skip` as an action, We want to extend the `action`
+functionality to support more usable options like `fs-backup` (File system backup) and `snapshot` (VolumeSnapshots).
+
+## Goals
+- Extending the VolumePolicies to support more actions like `fs-backup` (File system backup) and `snapshot` (VolumeSnapshots).
+- Improve user experience when backing up Volumes via Velero
+
+## Non-Goals
+- No changes to existing approaches to opt-in/opt-out annotations for volumes
+- No changes to existing `VolumePolicies` functionalities
+
+
+## Use-cases/Scenarios
+
+**Use-case 1:**
```

Comment:
We have a use case where we want to exclude all drivers by default and only include specific drivers for snapshotting. 
Can this be achieved with the current design?

Reply:
@rkasukurthi yes, driver is included in the existing selection mechanism for resource policies. This design reuses all of that and just adds new policy actions. See https://velero.io/docs/main/resource-filtering/#resource-policies

You'd probably use something like this:
```
- conditions:
    csi:
      driver: aws.ebs.csi.driver
  action:
    type: snapshot
- conditions:
  action:
    type: fs-backup
```
The above would snapshot ebs-csi volumes and use fs-backup for the rest -- if you want to skip completely any volumes not matching the specific drivers, then you'd use "skip" instead of "fs-backup".